---
title: "ctnnd2_qPCR_analysis"
output: html_document
date: "2024-06-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(xlsx)
library(stringr)
library(tidyverse)
library(ggplot2)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r Preprocessing}
d <- read.xlsx("2024-06-07_ctnnd2ko_gjb6_fabp7.xls", 3, header=F)

dd <- d[44:(nrow(d)-4),] #-4 rows at the bottom that are info about the run
print(dd[1,1]) #check first row
print(dd[nrow(dd),1]) #check last row

x <- d[43, 1:ncol(d)]
colnames(dd) <- c()
  for (i in 1:ncol(dd)){
    colnames(dd)[i] <- as.character(x[[i]])
  }

print(colnames(dd[15])) #check column number of CT data
colnames(dd)[15] <- 'CT'
colnames(dd)[16] <- 'Ct Mean'
colnames(dd)[17] <- 'Ct SD'

print(dd$CT)
sapply(dd$CT, class) # check class that your data has defaulted to

for (i in 1:nrow(dd)){
  if (dd$CT[i] == 'Undetermined'){
    dd$CT[i] <- 10e10
  }
}
print(dd$CT)

for (i in 1:nrow(dd)){
  if (dd$`Ct Mean`[i] == ''){
    dd$`Ct Mean`[i] <- 10e10
  }
}
print(dd$`Ct Mean`)
  
dd$CT <- as.numeric(dd$CT) # convert data to numeric class
dd$`Ct Mean` <- as.numeric(dd$`Ct Mean`)
dd$`Ct SD` <- as.numeric(dd$`Ct SD`)
print(dd$CT) 
sapply(dd$CT, class) # check success of conversion

print(dd$`Sample Name`)
dd$Genotype <- substr(dd$`Sample Name`,1,nchar(dd$`Sample Name`)-1)
sapply(dd$`Sample Name`, class) # if Sample name is not already "character" class, convert using code below
#dd$`Sample Name` <- as.character(levels(dd$`Sample Name`))
#dd$`Target Name` <- as.character(levels(dd$`Target Name`))[dd$`Target Name`]

dd <- dd[!dd$Task == 'NTC', ] # omit NTC

# omit if over SD threshold
dd$omit <- c()
for (i in 1:nrow(dd)){
  dd$omit[i] <- ifelse(dd$`Ct SD`[i] > 0.8, 'yes', 'no')
}
print(dd$omit)
# dd <- dd[!dd$omit == 'yes', ] # omit high SD
print(dd$omit)

Dat <- aggregate(dd, by=list(dd$`Ct Mean`, dd$`Sample Name`, dd$`Target Name`, dd$Genotype), mean)
Dat <- Dat[, 1:4]

names(Dat) <- c('CtMean', 'Sample', 'Target', 'Genotype')
print(Dat)


```

## Including Plots

You can also embed plots, for example:

```{r Analysis}
Dat$TargGroup <- factor(Dat$Target)
Dat$TargGroup <- as.numeric(Dat$TargGroup)

Dat$SampleGroup <- factor(Dat$Genotype)
Dat$SampleGroup <- as.numeric(Dat$SampleGroup)
Dat <- arrange(Dat, by_group = TargGroup)

# normalize to reference gene
k<-c()
for (i in 1:length(unique(Dat$TargGroup))){
  for (j in 1:length(unique(Dat$SampleGroup))){
    tmp <- Dat[Dat$SampleGroup == j, ]
    y <- tmp[tmp$TargGroup == i,1] - tmp[tmp$TargGroup == 1, 1] #enter housekeeping gene
    k <- c(k, y)
  }
}
Dat$dCt <- k
rm(k,y,tmp)

# normalize to WT
k<-c()
for (i in 1:length(unique(Dat$TargGroup))){
  for (j in 1:length(unique(Dat$SampleGroup))){
    tmp <- Dat[Dat$TargGroup == i, ]
    y <- tmp[tmp$SampleGroup == j, 'dCt'] - tmp[tmp$SampleGroup == 3, 'dCt'] #enter sample to normalize to
    k <- c(k, y)
  }
}
Dat$ddCt <- k
Dat$RQ <- 2^(-Dat$ddCt)

write.csv(Dat, '2024-06-07_ctnnd2ko_gjb6_fabp7-analyzed.csv', row.names=F)

#SampleGroup <- c(1:3)
#Genotype <- c('WT', 'HET', 'KO')
#ad <- data.frame(SampleGroup, Genotype)
#print(ad)

#Dat <- merge(Dat, ad, by='SampleGroup')
#print(Dat)

#TargGroup <- c(1:3)
#Exon <- c('B-actin', 'Exon 45', 'Exon 4-5', 'Exon 4-5', 'Exon 3-4', 'Exon 3-4', 'Exon 5')
#bd <- data.frame(TargGroup, Exon)
#print(bd)
#Dat <- merge(Dat, bd, by='TargGroup')
print(Dat)

#p1 <- ggplot(Dat, aes(x=Genotype,y=RQ)) +
p1 <- ggplot(Dat[Dat$TargGroup == 2,], aes(x=Genotype, fill = Genotype, y=RQ)) + 
  geom_bar(stat='identity', position="dodge") + 
  #ylim(0, 1.2) + geom_jitter() + 
  ylab('Fold Change Gjb6 / ActB') + 
  #theme_bw(base_size=9) + 
  ggtitle('Gjb6 expression at P21')
p1
ggsave(filename='gjb6_expression_p21.png')

p2 <- ggplot(Dat[Dat$TargGroup == 3,], aes(x=Genotype, fill = Genotype, y=RQ)) + 
  geom_bar(stat='identity', position="dodge") + 
  #ylim(0, 1.2) + geom_jitter() + 
  ylab('Fold Change Fabp7 / ActB') + 
  #theme_bw(base_size=9) + 
  ggtitle('Fabp7 expression at P21')
p2
ggsave(filename='fabp7_expression_p21.png')

print(Dat)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
